packages/myst-cli/src/process/site.ts

```diff
@@ -557,10 +610,12 @@ export async function processSite(session: ISession, opts?: ProcessSiteOptions):
   if (opts?.writeFiles ?? true) {
     await writeSiteManifest(session, opts);
     const states: ReferenceState[] = [];
+    const allPages: LocalProjectPage[] = [];
     await Promise.all(
       siteConfig.projects.map(async (project) => {
         if (!project.path) return;
         const { pages } = await loadProject(session, project.path);
+        allPages.push(...pages);
```

Comment:
This will be a non-deterministic order. Which we may want to avoid.

Reply:
I assume the ordering of `pages` is not guaranteed? OK, I'll impose an arbitrary sorting criterion.

Reply:
I meant something more like this:

```
const allPages: LocalProjectPage[] = (await Promise.all(
  ...
  ...
  return pages
)).flatten()
```

More the pattern that I am against, as I don't think it will make a difference now that I am looking closer.